### PR TITLE
Added 3 quizzes to 3.4 about containers and Docker

### PR DESCRIPTION
**Pull request by cfculler**
- Added the first quiz just below the container introduction and before Docker, so it focuses on the differences between containers and VMs
- Added second quiz just below Docker, before the exercises, and highlights some niche Docker concepts
- Added third quiz below the exercises that highlights the unique issue that arises in the exercises which stems from not having a volume attached. My goal for this quiz is to get them to consciously realize what the issue is why it happens. My hope is that this will immediately add value to the next section which covers compose and volumes

### CODE REVIEW

File: src/quizzes/chapter-3/3.4/containers-quiz.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const rawQuizdown = `
+---
+shuffleQuestions: true
+shuffleAnswers: true
+---
+
+# What are the key distinctions between containers and virtual machines in terms of resource utilization? Select all that apply
+
+- [x] Containers share the host operating system's kernel, while VMs run their own operating system kernel
+	> Good. This was the big revelation with containers that keeps them lightweight and easily scalable
```
Comment (jburns24) [inline src/quizzes/chapter-3/3.4/containers-quiz.js:10]: This is a great question but would like a note about how the shared kernel has security implications. From my understanding with VM's the machines are separated at the hardware level and which means a compromised VM does not endanger the other VMs. That is not true for containers, and compromising the kernel  of the container they can gain access to other containers.

File: src/quizzes/chapter-3/3.4/docker-quiz.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const rawQuizdown = `
+---
+shuffleAnswers: true
+---
+
+# Assume you made a Dockerfile that, when built into an image and ran, will install a handful of packages and log a success string. If you are on MacOS and your friend is on Windows, can you send them the image and expect the container to build and operate exactly the same?
```
Comment (jburns24) [inline src/quizzes/chapter-3/3.4/docker-quiz.js:6]:
```suggestion
# Assume you made a Dockerfile that, when built into an image and ran, will install a handful of packages and log a success string. If you are on MacOS and your friend is on Windows, can you send them the Dockerfile and expect the container to build and operate exactly the same?
```

File: src/quizzes/chapter-3/3.4/docker-quiz.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+1. [x] Yes
+	> Excellent, this is one of the main points of Docker and containers as a whole! You can share images between different environments and they *will* behave the same. They greatly help relieve the "It works on my machine" allegations
```
Comment (jburns24) [inline src/quizzes/chapter-3/3.4/docker-quiz.js:9]: _mostly_ I think you are right on the main point but different CPU architecture does throw a wrench in this. Like OOTB I do not think you will be able to run an ARM image on an x86 Windows box. The only reason we can run x86 images on MacOS is due to the Rosetta emulation layer.

Comment (jburns24) [reply src/quizzes/chapter-3/3.4/docker-quiz.js:9]: I think this is the case but again I think high level you are right and this is an edge case that they can learn about later

File: src/quizzes/chapter-3/3.4/containers-quiz.js (hunk 1 above)
Comment (jburns24) [reply src/quizzes/chapter-3/3.4/containers-quiz.js:10]: @cfculler

Comment (cfculler) [reply src/quizzes/chapter-3/3.4/containers-quiz.js]: Added this point you've made as feedback when they get the question right. To ensure apprentices really internalize this tradeoff I also added it as a deliverable talking point for a DOB review

**Review by jburns24 — APPROVED** [review]

**Pull request merged**